### PR TITLE
Adds days to keep field management job templates

### PR DESCRIPTION
**Pull request by acruzgon**
This PR closes [AAP-23137](https://issues.redhat.com/browse/AAP-23137)

It is adding a field for days to keep data only for `Cleanup Activity Stream` and `Cleanup Job Details` management job templates

### CODE REVIEW

**Review by RedOrNot — APPROVED** [review]
LG

File: frontend/awx/views/schedules/SchedulePage/ScheduleDetails.tsx [hunk 1]
```diff
@@ -58,6 +58,9 @@ export function ScheduleDetails() {
   if (!schedule) return <LoadingPage breadcrumbs tabs />;
   const isSystemJobTemplateSchedule: boolean = pathname.includes('management-jobs');
 
+  const isDays: boolean = JSON.stringify(schedule?.extra_data).includes('days');
```
Comment (AlexSCorey) [inline frontend/awx/views/schedules/SchedulePage/ScheduleDetails.tsx:61]:
```suggestion
  const hasDaysToKeep: boolean = JSON.stringify(schedule?.extra_data).includes('days');
```

File: frontend/awx/views/schedules/SchedulePage/ScheduleDetails.tsx [hunk 2]
```diff
@@ -102,12 +105,21 @@ export function ScheduleDetails() {
...
-            value={JSON.stringify(schedule.extra_data)}
-          />
-        </PageDetail>
+        {!isDays && (
+          <PageDetail fullWidth>
+            <PageDetailCodeEditor
+              label={t('Variables')}
+              value={JSON.stringify(schedule.extra_data)}
+            />
+          </PageDetail>
```
Comment (AlexSCorey) [inline frontend/awx/views/schedules/SchedulePage/ScheduleDetails.tsx:114]:
```suggestion
```

We should leave the CodeEditor to show only values related to survey's

File: frontend/awx/views/schedules/types.ts [hunk 3]
```diff
@@ -56,6 +56,7 @@ export interface ScheduleFormWizard {
   launch_config: LaunchConfiguration | null;
   prompt: PromptFormValues;
   schedule_days_to_keep: number;
+  daysToKeepData: number;
```
Comment (AlexSCorey) [inline frontend/awx/views/schedules/types.ts:59]: We should decide whether we want `daysToKeepData` or the line above `schedule_days_to_keep`

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
When navigating to schedules form from inventory sources the page crashes
Screenshot 2024-04-23 at 3 43 00 PM

File: frontend/awx/views/schedules/types.ts (hunk 3 above)
Comment (acruzgon) [reply frontend/awx/views/schedules/types.ts:59]: leaning more towards `schedule_days_to_keep`

**Review by AlexSCorey — APPROVED** [review]
Looks good to me.  Well done!

**Pull request merged**